docs/docs/prompts/optimize-prompt.mdx

```diff
@@ -0,0 +1,169 @@
+    },
+    {
+        "inputs": {
+            "question": "What is the value of $b$ if $5^b + 5^b + 5^b + 5^b + 5^b = 625^{(b-1)}$? Express your answer as a common fraction."
+        },
+        "expectations": {"answer": "\\frac{5}{3}"},
+    },
+    {
+        "inputs": {"question": "Evaluate the expression $a^3\\cdot a^2$ if $a= 5$."},
+        "expectations": {"answer": "3125"},
+    },
+    {
+        "inputs": {"question": "Evaluate $\\lceil 8.8 \\rceil+\\lceil -8.8 \\rceil$."},
+        "expectations": {"answer": "17"},
+    },
+]
+
+eval_data = [
+    {
+        "inputs": {
+            "question": "The sum of 27 consecutive positive integers is $3^7$. What is their median?"
+        },
+        "expectations": {"answer": "81"},
+    },
+    {
+        "inputs": {"question": "What is the value of $x$ if $x^2 - 10x + 25 = 0$?"},
+        "expectations": {"answer": "5"},
+    },
+    {
+        "inputs": {
+            "question": "If $a\\ast b = 2a+5b-ab$, what is the value of $3\\ast10$?"
+        },
+        "expectations": {"answer": "26"},
+    },
+    {
+        "inputs": {
+            "question": "Given that $-4$ is a solution to $x^2 + bx -36 = 0$, what is the value of $b$?"
+        },
+        "expectations": {"answer": "-5"},
+    },
+]
+
+# Optimize the prompt
+result = mlflow.genai.optimize_prompt(
+    target_llm_params=LLMParams(model_name="openai/gpt-4.1-mini"),
+    prompt=prompt,
+    train_data=train_data,
+    eval_data=eval_data,
+    scorers=[exact_match],
+    optimizer_config=OptimizerConfig(
+        num_instruction_candidates=8,
+        max_few_show_examples=2,
+    ),
+)
+
+# The optimized prompt is automatically registered as a new version
+print(result.prompt.uri)
+```
+
+After the optimization process is completed, you can visit the MLflow Prompt Registry page and see the optimized prompt.
+
+![Optimized Prompt](/images/llms/optimize-prompt/registered_prompt.png)
+
+Note that the optimized prompt of <APILink fn="mlflow.genai.optimize_prompt" /> expects the output to be a JSON string.
+Therefore, you need to parse the output using `json.loads` in your application.
+
+```python
+import mlflow
+import json
+import openai
+
+
+def predict(question: str, prompt_uri: str) -> str:
+    prompt = mlflow.load_prompt(prompt_uri)
+    content = prompt.format(question=question)
+    completion = openai.chat.completions.create(
+        model="gpt-4.1-mini",
+        messages=[{"role": "user", "content": content}],
+        temperature=0.1,
+    )
+
+    return json.loads(completion.choices[0].message.content)["answer"]
+```
+
+## Configuration
+
+You can customize the optimization process using `OptimizerConfig`, which includes the following parameters:
+
+- `num_instruction_candidates`: The number of candidate instructions to try.
+- `max_few_show_examples`: The maximum number of examples to show in few-shot demonstrations.
+- `optimizer_llm`: The LLM to use for optimization.
```

Comment:
Missing num_threads and algorithm?

Reply:
It's intentional actually. I think we just need to describe some of the main parameters here.